src/symbol/projection.js

```diff
@@ -260,13 +280,22 @@ function requiresOrientationChange(writingMode, firstPoint, lastPoint, aspectRat
             return {useVertical: true};
         }
     }
-
-    if (writingMode === WritingMode.vertical ? firstPoint.y < lastPoint.y : firstPoint.x > lastPoint.x) {
-        // Includes "horizontalOnly" case for labels without vertical glyphs
-        return {needsFlipping: true};
+    // Check if flipping is required for "verticalOnly" case.
+    if (symbol.writingMode === WritingMode.vertical) {
+        return (firstPoint.y < lastPoint.y) ? {needsFlipping: true} : null;
+    }
+    // Check if flipping is required for "horizontal" case.
+    const flipRequired = (firstPoint.x > lastPoint.x);
+    // If flipping is required, but the glyphs are lying roughly within the flip stage retain range,
+    // then first check if the placedSymbol has entered into this retain range before or not. if it
+    // has entered before, it should have a valid flip stage indicating the flip decision made by the
+    // the first time entering, take the flip stage directly. Otherwise, it should be the first time
+    // entering, take the original flip decision and update placeSymbol's flip stage in this range.
```

Comment:
The comment might need updating to reflect the change. I think I'd describe the subtle difference in logic as: `needsFlipping` stores the flip decision from the previous frame, and that decision is reused when the symbol is in the retain range.